File: C/2003.c
#include <stdio.h>  // 수들의 합 2


int main()
{
	int n, m;
	scanf("%d %d", &n, &m);

	int arr[10001] = { 0, };
	for (int i = 0; i < n; i++)
	{
		scanf("%d", &arr[i]);
	}

	int cnt = 0;
	for (int i = 0; i < n; i++)
	{
		int j = i + 1;
		int sum = arr[i];
		while (1)
		{
			if (sum == m)
			{
				cnt++;
				break;
			}
			else if (sum > m)
			{
				break;
			}

			if (j == n)
			{
				break;
			}

			sum += arr[j];
			j++;
		}
	}
	printf("%d", cnt);

	return 0;
}

File: JAVA/src/_2003.java
import java.util.Scanner;

// 수들의 합 2
public class _2003 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int M = sc.nextInt();

        int[] arr = new int[N + 1];
        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }

        int sum = 0;
        int start = 0, end = 0;
        int cnt = 0;
        while (true) {
            if (sum >= M) {
                sum -= arr[start++];
            }
            else if (end == N) {
                break;
            }
            else {
                sum += arr[end++];
            }

            if (sum == M) {
                cnt++;
            }
        }

        System.out.println(cnt);
    }
}
